src/wallet/db.h

```diff
@@ -35,10 +35,11 @@ class CDBEnv
     void EnvShutdown();
 
 public:
-    mutable CCriticalSection cs_db;
+    std::recursive_mutex cs_db;
```

Comment:
In commit "Replace cs_db with a recursive mutex use with a conditional_variable_any"

cs_db is actually already a recursive mutex (it inherits from std::recursive_mutex). I think you could just leave it unchanged, and continue using the LOCK macro instead of std::lock_guard everywhere. This would make the commit smaller & simpler.

Reply:
I tried that, but it kept giving me a compiler error.

Reply:
https://github.com/bitcoin/bitcoin/pull/12493#discussion_r169655317

> I tried that, but it kept giving me a compiler error.

What is the compiler error? The following compiles for me: c271de2d62881c0880bf64b830d72f686967bbfb (fetchable with `git fetch https://github.com/ryanofsky/bitcoin pr/lreload`).

I didn't do any real testing with it but it passes python & unit tests.

Reply:
Ah, I was doing something with the macros and CCriticalSection. It was an error about not having an `unlock` method.

I'll test this out, but from testing I just did with the current code, it looks like it does work, so this should work too.

Reply:
@ryanofsky I've used your commit.